utils/parsing_utils.py

```diff
@@ -1,40 +1,43 @@
 import glob
 import os
+from pathlib import Path
 
-def remove_extension(input):
-
-    filename = input.split("/")[-1].split(".")[0]
 
-    if len(filename) > 2:
-        filename = ".".join(filename[0:-1])
-    else:
-        filename = input.split("/")[-1].split(".")[0]
+def remove_extension(input):
+    """Removes filename extension AND directory path"""
+    filename = Path(input)
+    filename = filename.with_suffix("")
+    filename = os.path.basename(filename)
+    return str(filename)
```

Comment:
`os.path.basename` returns a string. No need to call `str` here.

Reply:
agreed. It just lingered there because I first didn't have the basename line, as I hadn't realised the function was supposed to do more than just remove the extension